py/desispec/scripts/proc.py

```diff
@@ -722,26 +731,61 @@ def main(args=None, comm=None):
             inputs = comm.bcast(inputs, root=0)
             outputs = comm.bcast(outputs, root=0)
 
-            #- split communicator by 20 (number of bundles)
-            extract_size = 20
-            if (rank == 0) and (size%extract_size != 0):
-                log.warning('MPI size={} should be evenly divisible by {}'.format(
-                    size, extract_size))
+            run_extraction = True
+
+            if args.gpuextract and len(cmds) > 0:
+                #- GPU extraction currently assumes single node extraction group
+                #- If extract size is not provided, use 1 rank per GPU available plus 2 ranks for IO
+                #- On Perlmutter GPU, --extract-size 22 provides the best performance.
+                #- 22 ranks = 5 ranks per GPU * 4 GPUs + 2 ranks for IO
+                extract_size = args.extract_size
+                if extract_size is None:
+                    import cupy as cp
+                    ngpus = cp.cuda.runtime.getDeviceCount()
+                    if rank == 0:
+                        log.info(f"{rank} found {ngpus} gpus")
+                    extract_ranks_per_gpu = 1
+                    extract_ranks_io = 2
+                    extract_size = extract_ranks_io + ngpus * extract_ranks_per_gpu
+                extract_group = 0
+                num_extract_groups = 1
+                extract_ranks = list(range(extract_size))
+                #- Create a comm group for the ranks that will perform extraction
+                if rank in extract_ranks:
+                    extract_incl = comm.group.Incl(extract_ranks)
+                    comm_extract = comm.Create_group(extract_incl)
+                    from gpu_specter.mpi import ParallelIOCoordinator
+                    coordinator = ParallelIOCoordinator(comm_extract)
+                else:
+                    run_extraction=False
+            else:
```

Comment:
There should be another case here for setting up the communicator for the CPU code path of gpu_speceter (when `--gpuspecter` is specified without `--gpuextract`.

Reply:
Would the additional case be for any other reason than to [set extract_size = 16](https://github.com/desihub/desispec/blob/dmargala-scale-run-wip/py/desispec/scripts/proc.py#L780) instead of 20?

Reply:
`extract_size = 16` should be a decent default for the cpu version of gpu_specter on cori haswell and perlmutter gpu. It does look like it be okay to share the existing cpu-path logic for comm_extract although it needs `extract_size = args.extract_size`.

Reply:
I think it is best to set extract_size = 20, 16, and 22 for specter, gpu_specter_cpu and gpu_specter_gpu, respectively, when the `--extract-size` flag is not used, and to the value specified otherwise.  

I will wait to get feedback before making this change.